src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,788 @@
+        """Gets the dwell time of each pixel based on Channels, samples, delay and oversampling.
+
+        Args:
+            OS (int): oversampling rate
+            CH (int): number of channels
+            samples (int): number of samples per channel
+            delay (float): delay between samples
+            TRIGGER (bool): trigger state
+
+        Returns:
+            int: dwell time in us
+
+        """
+        delayCycle = functions.delayCycles(delay)  # delay in cycles
+        dt = functions.dwellTime(self.oversampling.value, self.numberOfChannels.value, delayCycle, trigger) * 10e-6
+        return dt
+
+    def get_scan_time(self, dwell_time: float, points_fast: int, points_slow: int) -> float:
+        """Gets the scan time of the system based on dwell time and number of points.
+
+        Args:
+            dwell_time (int): dwell time in us
+            points_fast (int): number of points in the fast axis default is x axis
+            points_slow (int): number of points in the slow axis default is y axis
+        Returns:
+            float: scan time in seconds
+        """
+        return functions.scanTime(dwell_time, points_fast, points_slow)
+
+    async def start_scan(self, simulate=False) -> None:
+        """
+        Starts a scan with the EBIC scan controlbox.
+        Send settings to the server/client to initiate scan.
+        """
+        self._dwell_time = self.get_dwell_time(self.delay)
+        self._scan_time = self.get_scan_time(self._dwell_time, self.repetition.value[0], self.repetition.value[1])
+
+        result = await self.StartScanOPCUA.call_method(
+            ua.NodeId("ScanObjectOPCUA", self.idx),  # FIXME this might not be the right format
+            ua.QualifiedName("StartScanOPCUA", self.idx),
+            self.repetition.value[0],
+            self.repetition.value[1],
+            self.oversampling.value,
+            self.numberOfChannels.value,
+            self.spp.value,
+            self.delay,
+            TRIGGER=True,
+            Simulate=simulate)
+
+    def stop_scan(self):
+        # this function is to be used in the special SPARC acq
+        pass
+
+    async def retrieve_acquired_data(self):
+        """
+        Reads a variable from the server.
+        """
+        data = await self._client.nodes.root.get_child(["0:Objects", f"{self.idx}:{DATA_VAR_NAME}"])
+        value = await data.read_value()
+        return numpy.array(list(value), dtype=numpy.float64)
+
+    async def read_variable(self, obj_name: str, var_name: str) -> Any:
+        """
+        Reads a variable from the OPC server.
+        :param obj_name:
+        :param var_name:
+        :return:
+        """
+        var = await self._client.nodes.root.get_child(f"0:Objects/{self.idx}:{obj_name}/{self.idx}:{var_name}")
+        value = await var.read_value()
+        return value
+
+    def on_repetition_change(self, value):
+        return value
+
+    def on_spp_change(self, value):
+        return value
```

Comment:
My understanding is that the dwell time is defined by the samples per pixel, oversampling, and delay. However, in Odemis, typically we have just one setting the "dwellTime". Here, the driver *must* provide a dwellTime VA so that the acquisition code can copy the dwellTime of the standard e-beam scanner and let the EBIC know (and the closest dwellTime < the requested dwellTime should be used). 

Ideally, we would compute the best 3 settings based on the dwellTime. I don't know if that's possible. If not, we could allow the user to adjust the samples per pixel, oversampling, and the driver would pick the best delay for matching the dwellTime.